src/libecalc/presentation/exporter/infrastructure.py

```diff
@@ -0,0 +1,405 @@
+from datetime import datetime
+from typing import Iterable, List, Optional, Tuple
+
+from typing_extensions import assert_never
+
+from libecalc.application.graph_result import GraphResult
+from libecalc.common.component_type import ComponentType
+from libecalc.common.errors.exceptions import ProgrammingError
+from libecalc.common.logger import logger
+from libecalc.common.temporal_model import TemporalModel
+from libecalc.common.time_utils import Frequency, Period, Periods
+from libecalc.common.units import Unit
+from libecalc.common.utils.rates import TimeSeries, TimeSeriesFloat, TimeSeriesRate, TimeSeriesStreamDayRate
+from libecalc.core.result import GeneratorSetResult
+from libecalc.dto import FuelConsumer, GeneratorSet
+from libecalc.dto.utils.validators import convert_expression
+from libecalc.presentation.exporter.domain.exportable import (
+    Attribute,
+    AttributeMeta,
+    AttributeSet,
+    Exportable,
+    ExportableSet,
+    ExportableType,
+)
+from libecalc.presentation.yaml.yaml_types.emitters.yaml_venting_emitter import (
+    YamlDirectTypeEmitter,
+    YamlOilTypeEmitter,
+    YamlVentingType,
+)
+
+
+class TimeSeriesAttribute(Attribute):
```

Comment:
I like this idea, and it makes sense to divide query into this imo. I cant help but thinking that this could be useful outside LTP/STP as well, perhaps if it was  a bit more generic/less directly related to LTP/STP? I dont see all the details, but a "usefuel" layer imo

Reply:
Not sure I understand. I agree that an interface for TimeSeriesAttributes makes sense outside exporter. Currently we are using TimeSeries classes directly (in common, some places we use pydantic functionality, other places we don't). 

I also think the Exportable interface makes sense in an energy usage context, not only for export. But it is a bit specific. Instead of `get_power_consumption` returning an AttributeSet, it might make more sense to have `get_power_consumers` where each power_consumer can answer `get_power_consumption -> Attribute/AttributeSet`. Since we might be interested in other stuff also.

Reply:
I agree that it is too specific the way it is now, but the "notion" of splitting into "energy producing/consuming infrastructure" - as discussed.